src/script_lua.c

```diff
@@ -816,6 +845,36 @@ static int luaRedisGenericCommand(lua_State *lua, int raise_error) {
     return 1;
 }
 
+/* Our implementation to lua pcall.
+ * We need this implementation for backward
+ * comparability with older Redis versions.
+ *
+ * On Redis 7, the error object is a table,
+ * compare to older version where the error
+ * object is a string. To keep backward
+ * comparability we catch the table object
+ * and just return the error message. */
+static int luaRedisPcall(lua_State *lua) {
+    int argc = lua_gettop(lua);
+    int result = 1;
+    if (lua_pcall(lua, argc - 1, LUA_MULTRET, 0)) {
+        /* Error */
+        result = 0;
+        if (lua_istable(lua, -1)) {
+            lua_getfield(lua, -1, "err");
```

Comment:
Verify if `lua_checkstack` is needed here

Reply:
Verified here: https://github.com/redis/redis/pull/10329/commits/c297dc6f6aa881aa48a1a423806bcc326fbdf321